### ISSUE TITLE
PanelMenu: CTRL+Click does not open new Tab (Firefox)

### ISSUE THREAD

**Issue by cyanotyp** [primefaces/primefaces#5265]
## 1) Environment
- PrimeFaces version: 7.0
- Does it work on the newest released PrimeFaces version?  No 
- Does it work on the newest sources in GitHub?  No, Version? 7.1 snapshot
- Application server + version: Oracle WLS 12.2.1.3.0
- Affected browsers: Mozilla Firefox 60ESR, 68ESR

## 2) Expected behavior

Do a CTRL+Click on MenuItem - and Link opens in **NEW** Tab.

## 3) Actual behavior

We recently upgraded from PF5.3 to 7.0 and came across a strange behaviour in our navigation using PanelMenu (with SubMenu and MenuItems)
Our MenuItems look like
<p:menuitem id="menuLinkXXX" value="label" immediate="true" outcome="/outcomepath" />

In PF5.3 it is possible to do a CTRL+Click on these MenuItems - and they open in **NEW** Tab.
In PF7.0, CTRL+Click opens the Link in the **SAME** tab now. 

Work-Around is to use middle mouse key or RMK - open in new Tab, which is not very handy.

Issue is reproducible on PF Showcase running PF 7.1-snapshot 

PrimeFaces Forums topic:
https://forum.primefaces.org/viewtopic.php?f=3&t=60077

## 4) Steps to reproduce

Open Menuitem "Navigations" - "Links" - "Primefaces " in the showcase page
https://www.primefaces.org/showcase/ui/menu/panelMenu.xhtml
CTRL+Click on "Home" -> Link opens in the **SAME** Tab

## 5) Sample XHTML

https://www.primefaces.org/showcase/ui/menu/panelMenu.xhtml

## 6) Sample bean

n/a

**Comment by tandraschko** [comment]
No idea but we just render a link here - so i should be up to the browser?

**Comment by melloware** [comment]
@cyanotyp Yes can you try any other link in Firefox not in a PF app?  Maybe Firefox disabled this behavior?

**Comment by melloware** [comment]
Ahh I can reproduce.  Investigating....

**Comment by melloware** [comment]
I think its definitely a firefox issue: https://support.mozilla.org/en-US/questions/965091

**Comment by melloware** [comment]
Found it.  PanelMenu is doing this...

```javascript
.click(function(e) {
            var currentLink = $(this);
            $this.focusItem(currentLink.closest('.ui-menuitem'));

            var href = currentLink.attr('href');
            if(href && href !== '#') {
                window.location.href = href;
            }
            e.preventDefault();
        });
```

its handling the link click itself.

**Comment by tandraschko** [comment]
Why are we doing this? Because of the focus?

**Comment by melloware** [comment]
Yeah it looks like a bunch of the menu's do this.  I am doing a PR now for your review.

**Comment by melloware** [comment]
@cyanotyp if you need an immediate fix just include this script on your page.

```javascript
PrimeFaces.widget.PanelMenu.prototype.bindEvents = function() {
        var $this = this;

        this.headers.mouseover(function() {
            var element = $(this);
            if(!element.hasClass('ui-state-active')) {
                element.addClass('ui-state-hover');
            }
        }).mouseout(function() {
            var element = $(this);
            if(!element.hasClass('ui-state-active')) {
                element.removeClass('ui-state-hover');
            }
        }).click(function(e) {
            var header = $(this);

            if (!$this.cfg.multiple) {
                $this.collapseActiveSibling(header);
            }

            if (header.hasClass('ui-state-active'))
                $this.collapseRootSubmenu($(this));
            else
                $this.expandRootSubmenu($(this), false);

            $this.removeFocusedItem();
            header.focus();
            e.preventDefault();
        });

        this.menuitemLinks.mouseover(function() {
            $(this).addClass('ui-state-hover');
        }).mouseout(function() {
            $(this).removeClass('ui-state-hover');
        }).click(function(e) {
            var currentLink = $(this);
            $this.focusItem(currentLink.closest('.ui-menuitem'));
            var href = currentLink.attr('href');
            if(href && href !== '#') {
                if (e.ctrlKey) {
                    window.open(href, '_blank').focus();
                } else {
                    window.location.href = href;
                }
            }
            e.preventDefault();
        });

        this.treeLinks.click(function(e) {
            var link = $(this),
            submenu = link.parent(),
            submenuList = link.next();

            if(submenuList.is(':visible'))
                $this.collapseTreeItem(submenu);
            else
                $this.expandTreeItem(submenu, false);

            e.preventDefault();
        });

        this.bindKeyEvents();
    };
```

**Comment by cyanotyp** [comment]
Hello all,

thank you for the quick updates and hotfix !
I will try to implement the same tomorrow and post an update here :)

**Comment by cyanotyp** [comment]
Hello @melloware 
your hotfix works perfect for our project, our users will be very happy !

**Comment by melloware** [comment]
No problem!  This will be fixed for good in 7.1  Thanks for reporting the issue!

**Comment by ravihariharan** [comment]
@mertsincan  This fix of menu are also required to be done also on the premium templates.  The primefaces showcase works with CTRL+Click , but the https://www.primefaces.org/poseidon/ has the same issue as described above.
